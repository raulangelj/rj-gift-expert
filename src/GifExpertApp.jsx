import { useState } from "react"
import { GifGrid, AddCategory } from "./components"

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Friends', 'Iron Man'])

  const onAddCategory = ( newCategory ) => {
    if (categories.includes(newCategory)) return
    setCategories((oldState) => [newCategory, ...oldState])
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory
        setCategories={setCategories}
        onNewCategory={ onAddCategory }
      />
      {
        categories.map(category => (
          <GifGrid
            key={category}
            category={category}
          />
        ))
      }
    </>
  )
}

export default GifExpertApp