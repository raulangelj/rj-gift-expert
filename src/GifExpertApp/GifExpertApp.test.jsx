import { render } from "@testing-library/react"
import { GifExpertApp } from "./GifExpertApp"

describe('Prueba en <GifExpertApp />: ', () => {
  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<GifExpertApp />)
    expect(container).toMatchSnapshot()
  })
})