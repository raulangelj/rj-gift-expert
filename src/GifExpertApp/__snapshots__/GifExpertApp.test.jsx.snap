// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifExpertApp />:  Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Friends
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Iron Man
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
